### DISCUSSION TITLE
Deployment build error with Astro project

### DISCUSSION THREAD

**Discussion by hodeinavarro** [vercel/vercel#8339] [Help] [answered]
Hello,

I have a project that follows the getting started / basic guides of [docs.astro.build](https://docs.astro.build) and when I try to deploy on Vercel it fails with the following error:

```
node_modules/.pnpm/astro@1.0.0-rc.7/node_modules/astro/client-base.d.ts(1,23): error TS2688: Cannot find type definition file for 'vite/types/importmeta'.
```

This is working and building also works on local, I could also find that the mentioned file existed in the node_modules folder.

Dependencies:
```json
"devDependencies": {
  "@astrojs/tailwind": "^0.2.5",
  "@tailwindcss/aspect-ratio": "^0.4.0",
  "@tailwindcss/line-clamp": "^0.4.0",
  "@tailwindcss/typography": "^0.5.4",
  "astro": "^1.0.0-rc.7",
  "autoprefixer": "^10.4.8",
  "postcss": "^8.4.14",
  "prettier": "^2.7.1",
  "prettier-plugin-astro": "^0.3.0",
  "tailwindcss": "^3.1.8"
}
```

I have been searching internet, issues, etc. and found that it could be a typescript thing, I have a ```tmp.ts``` file under ```src``` exporting an empty object, and here is my ```tsconfig.json``` following the guides provided by Astro.

```json
{
	"compilerOptions": {
		"target": "ESNext",
		"module": "ESNext",
		"moduleResolution": "node",
		"resolveJsonModule": true,
		"isolatedModules": true,
		"types": ["astro/client"],
		"outDir": "./dist",
		"importsNotUsedAsValues": "error",
		"baseUrl": ".",
		"paths": {
			"@/components/*": ["src/components/*"],
			"@/layouts/*": ["src/layouts/*"]
		}
	}
}
```

The project is an ```index.astro``` using a ```Landing.astro``` component using a layout ```Layout.astro``` that as a nav with ```NavItem.astro```. Nothing fancy just pure "astro language" and "astro language imports"

astro.config.mjs
```typescript
import tailwind from '@astrojs/tailwind'
import { defineConfig } from 'astro/config'

export default defineConfig({
	integrations: [tailwind()],
})

```

pages/index.astro
```astro
---
import Landing from '@/components/Landing.astro';
import Layout from "@/layouts/Layout.astro";
---
<Layout title="My Astro App">
    <Landing />
</Layout>

```

layouts/Layout.astro
```astro
---
import NavItem from '@/components/NavItem.astro';

export interface Props {
	title: string;
}

const { title } = Astro.props as Props;
---
<style>
	/** Styling things here */
</style>
<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta name="viewport" content="width=device-width" />
		<title>{title}</title>
	</head>
	<body>
		<nav class="flex justify-center items-center p-4 gap-5 flex-wrap">
			<NavItem text="NavItem" href='http://127.0.0.1:3000' />
		</nav>
		<slot />
	</body>
</html>
```

components/NavItem.astro
```astro
---
export interface Props {
	text: string;
	href: string;
}

const {  text, href } = Astro.props as Props;
---
<a class='font-extralight tracking-widest text-base hover:bg-gradient-to-br hover:from-cyan-500 hover:via-purple-500 hover:to-rose-500 hover:bg-clip-text hover:text-transparent hover:animate-text-gradient-fast' href={href}>{text}</a>

```

components/Landing.astro
```astro
---

---
<section class="p-4 text-center h-full flex flex-col justify-center items-center gap-4">
    <div class="flex flex-col bg-gradient-to-br from-cyan-500 via-purple-500 to-rose-500 bg-clip-text text-transparent animate-text-gradient">
        <h1 class="text-5xl">Page Title</h1>
        <h2 class="text-xl">Page Subtitle</h2>
    </div>
</section>

```

I have changed nothing on the Vercel deployment configuration, the framework preset is ```Astro``` as detected, root is ```./```, Node.js is ```16.x```.

Thank for any help provided, much needed ❤️

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @hodeinavarro! I see you're using `"astro": "^1.0.0-rc.7"` in your `package.json` which may end up being a slightly different version when the packages are freshly installed on a new build. Does it work if you take out the `^` to pin `1.0.0-rc.7` specifically?

I appreciate all the code examples you shared! But I'm not quite seeing how they all fit together in your app 😅 (I have not yet used Astro, so please forgive my ignorance). If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or the repo you're using, I would be happy to take a closer look.

**Reply by hodeinavarro** [reply]
I'll try to check out your solution when I can, thank you for the response @amyegan 

I've been testing some more and the build fails even on the "getting started" because of ```tsc --noEmit``` that I had on my scripts.
The script was as mentioned on [their docs](https://docs.astro.build/en/guides/typescript/#type-checking):
```package.json
"scripts": {
    "build": "astro check && tsc --noEmit && astro build",
},
```

I believe this could be on Vercels end since the declaration ```tsc``` references (vite/types/importmeta) exists, but maybe we could also open an issue there.
